sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/BufferSendState.scala

```diff
@@ -185,11 +188,32 @@ class BufferSendState(
           hostBounceBuffer.buffer
         }
 
-        acquiredBuffs.foreach { case RangeBuffer(blockRange, rapidsBuffer) =>
-          require(blockRange.rangeSize() <= bounceBuffToUse.getLength - buffOffset)
-          rapidsBuffer.copyToMemoryBuffer(blockRange.rangeStart, bounceBuffToUse, buffOffset,
-            blockRange.rangeSize(), serverStream)
-          buffOffset += blockRange.rangeSize()
+        // `copyToMemoryBuffer` can throw if the `RapidsBuffer` is in the DISK tier and
+        // the file fails to mmap. We catch the `IOException` and attempt a retry
+        // in the server.
+        var needsCleanup = false
+        try {
+          acquiredBuffs.foreach { case RangeBuffer(blockRange, rapidsBuffer) =>
+            needsCleanup = true
+            require(blockRange.rangeSize() <= bounceBuffToUse.getLength - buffOffset)
+            rapidsBuffer.copyToMemoryBuffer(blockRange.rangeStart, bounceBuffToUse, buffOffset,
+              blockRange.rangeSize(), serverStream)
+            buffOffset += blockRange.rangeSize()
+          }
+          needsCleanup = false
+        } catch {
+          case ioe: IOException =>
+            val ex = new RapidsShuffleSendPrepareException(
+              s"Error while copying to bounce buffer for executor ${peerExecutorId} and " +
+                  s"header ${TransportUtils.toHex(peerBufferReceiveHeader)}")
+            ex.addSuppressed(ioe)
```

Comment:
Why is this a suppressed exception instead of a caused exception?  Suppressed exceptions are ones that occur while processing another exception, but the RapidsShuffleSendPrepareException is caused by the IOException.